rust/core-lib/src/file.rs

```diff
@@ -161,29 +172,30 @@ impl FileManager {
         } else {
             let encoding = self.file_info[&id].encoding;
             try_save(path, text, encoding).map_err(|e| FileError::Io(e, path.to_owned()))?;
-            self.file_info.get_mut(&id).unwrap().mod_time = get_mod_time(path);
+            if let Some(v) = self.file_info.get_mut(&id) {
+                v.mod_time = get_mod_time(path);
+                #[cfg(feature = "notify")]
+                update_current_position_in_tail(v, text);
+            }
         }
         Ok(())
     }
-}
-
-fn try_load_file<P>(path: P) -> Result<(Rope, FileInfo), FileError>
-where
-    P: AsRef<Path>,
-{
-    // TODO: support for non-utf8
-    // it's arguable that the rope crate should have file loading functionality
-    let mut f =
-        File::open(path.as_ref()).map_err(|e| FileError::Io(e, path.as_ref().to_owned()))?;
-    let mod_time =
-        f.metadata().map_err(|e| FileError::Io(e, path.as_ref().to_owned()))?.modified().ok();
-    let mut bytes = Vec::new();
-    f.read_to_end(&mut bytes).map_err(|e| FileError::Io(e, path.as_ref().to_owned()))?;
 
-    let encoding = CharacterEncoding::guess(&bytes);
-    let rope = try_decode(bytes, encoding, path.as_ref())?;
-    let info = FileInfo { encoding, mod_time, path: path.as_ref().to_owned(), has_changed: false };
-    Ok((rope, info))
+    #[cfg(feature = "notify")]
+    pub fn toggle_tail(&mut self, id: BufferId, enabled: bool) -> Result<(), FileError> {
```

Comment:
As a public function, this should have at least a brief bit of documentation.

Reply:
I agree. I will add comments.